packages/network-controller/src/NetworkController.ts

```diff
@@ -398,67 +480,38 @@ export class NetworkController extends BaseControllerV2<
         EIPS: {},
       };
     });
-    const { rpcUrl, type, chainId } = this.state.providerConfig;
-    this.#configureProvider(type, rpcUrl, chainId);
+    this.#applyNetworkSelection();
     this.messagingSystem.publish('NetworkController:networkDidChange');
     await this.lookupNetwork();
   }
 
-  #registerProvider() {
-    const { provider } = this.getProviderAndBlockTracker();
-
-    if (provider) {
-      this.#ethQuery = new EthQuery(provider);
-    }
-  }
-
-  #setupInfuraProvider(type: InfuraNetworkType) {
```

Comment:
The code in this and the next method is now split up into two methods. The code to create the network client is now in `#createAutoManagedNetworkClientRegistry` and the code to update the provider is now in `#selectNetwork`.